### ISSUE TITLE
Blazor doesn't seem to follow default namespace

### ISSUE THREAD

**Issue by ErikApption** [aspnet/AspNetCore#7902]
### Describe the bug
Blazor generates cs files with a namespace using the project name instead of the default namespace
My csproj has
```xml
  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
    <OutputType>Exe</OutputType>
    <LangVersion>7.3</LangVersion>

    <!-- The linker is not beneficial for server-side execution. Re-enable this if you switch to WebAssembly execution. -->
    <BlazorLinkOnBuild>false</BlazorLinkOnBuild>
    <Configurations>Debug;Release</Configurations>
    <RootNamespace>WebAppBootStrap.App</RootNamespace>
  </PropertyGroup>
```

but the generated WebAppBootStrap.App-core3\_ViewImports.cshtml has

namespace WebAppBootStrap.App_core3 instead of WebAppBootStrap.App

### To Reproduce
1. Create a project WebAppBootStrap.App-core3 
2. Set WebAppBootStrap.App as default namespace
3. Write startup.cs
 as 
```cs
namespace WebAppBootStrap.App
{
    public class Startup
    {
        public void ConfigureServices(IServiceCollection services)
        {
            services.AddSingleton<AppState>();
        }

        public void Configure(IComponentsApplicationBuilder app)
        {
            app.AddComponent<App>("app");
        }
    }
}
```

**Comment by josephayoung** [comment]
Possibly related: 

If the project name doesn't match the project _folder_ name, none of the in-project components will be detected as components during the `BuildRenderTree` generation. They get picked up only as normal tags.

Ex.: A new project is created using the Razor Components template. The client app project file is renamed from `Project1.Client.App.csproj` to `Project1.App.csproj`, but the project folder isn't renamed. (In VS2019, project folders are out of view, and aren't renamed automatically when a project is renamed).

`Component2.cshtml` is located in the same folder as `Component1.cshtml`. `<Component2 />` is used inside of `Component1.cshtml`. 

When the project is built, both component files are detected as components and generated class files are produced for both files. Generated namespaces match the folder structure.

However, in `Component1.cshtml.g.cs`, the code corresponding to the `Component2 />` tag is `builder.OpenElement(0, "Component2")` rather than `builder.OpenComponent<Project1.App.Shared.Component2>(16)`.

**Comment by josephayoung** [comment]
#5524

**Comment by nvmkpk** [comment]
The project/folder name issue is reporeted here: #6155

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us, @ErikApption.
This will be fixed in VS 16.1 release.

**Comment by isc30** [comment]
Is there any way to specify the desired namespace for the component?
